### PR TITLE
Build installers at the end of the repo build and allow building with desktop msbuild

### LINKED ISSUE #58984: -BuildManaged -BuildInstallers doesn't work as installer build too early
When building installers without first building the rest of the repository (managed + native) in a previous build execution, the build fails as the installer build runs too early. This is probably not a problem as that's not a common workflow but with the VMR we need to be able to build the entire repository in one go.

Blocks https://github.com/dotnet/sdk/pull/44828?new_mergebox=true

The solution that I'm thinking of is adding a Traversal project that builds the installer projects and adding that to eng/Build.props at the very end with a `BuildInParallel=false` metadata so that it runs after everything else. The installer project will still build in parallel.

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/aspnetcore/issues/58984
Unblocks https://github.com/dotnet/sdk/pull/44828 (removes the patch)
Depends on https://github.com/dotnet/aspnetcore/pull/59150

- Build the Installer projects after everything else got built by using Arcade's new `BuildStep` metadata.
- Allow building the entire repository from a single build invocation.
- When using desktop msbuild, build the entire repository from a single build invocation.
- Condition SharedFrameworkBundle.wixproj correctly when building inside the VMR.

# {PR title}

- [ ] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [ ] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

Summary of the changes (Less than 80 chars)

## Description

{Detail}

Fixes #{bug number} (in this specific format)

### CODE REVIEW

**Comment by ViktorHofer** [comment]
Adding the NO MERGE label until I fully tested the changes inside the VMR

**Comment by wtgodbe** [comment]
Local Development job failures are unrelated

**Comment by ViktorHofer** [comment]
@ellahathaway this will fix the Sign and Publish issue as well inside the VMR.

**Comment by ViktorHofer** [comment]
Needs https://github.com/dotnet/aspnetcore/pull/59150

File: eng/Build.props [hunk 1]
```diff
@@ -269,9 +211,83 @@
...
+        <InstallerProject Include="$(RepoRoot)src\Installers\Windows\AspNetCoreModule-Setup\ANCMV2\ANCMV2.wixproj" AdditionalProperties="Platform=$(_WixTargetPlatform)" />
+        <!-- Build the targeting pack installers -->
+        <InstallerProject Include="$(RepoRoot)src\Installers\Windows\TargetingPack\TargetingPack.wixproj" AdditionalProperties="Platform=$(_WixTargetPlatform)" />
+        <!-- Build the SharedFramework installers -->
+        <InstallerProject Include="$(RepoRoot)src\Installers\Windows\SharedFrameworkBundle\SharedFrameworkBundle.wixproj" AdditionalProperties="Platform=$(_WixTargetPlatform)" />
+        <!-- Build the SharedFramework wixlib -->
+        <InstallerProject Include="$(RepoRoot)src\Installers\Windows\SharedFrameworkLib\SharedFrameworkLib.wixproj" AdditionalProperties="Platform=$(_WixTargetPlatform)" />
+      </ItemGroup>
+
+      <ItemGroup Condition="'$(DotNetBuild)' == 'true' and ('$(DotNetBuildPass)' == '2') and and '$(TargetOsName)' == 'win' and '$(TargetArchitecture)' == 'x64'">
```
Comment (wtgodbe) [inline eng/Build.props:277]:
```suggestion
      <ItemGroup Condition="'$(DotNetBuild)' == 'true' and ('$(DotNetBuildPass)' == '2') and '$(TargetOsName)' == 'win' and '$(TargetArchitecture)' == 'x64'">
```

File: eng/Npm.Workspace.FunctionalTests.nodeproj [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<Project>
+<Project DefaultTargets="Build">
```
Comment (wtgodbe) [inline eng/Npm.Workspace.FunctionalTests.nodeproj:1]: What made this necessary?

File: eng/Build.props (hunk 1 above)
Comment (ViktorHofer) [reply eng/Build.props:277]: Oh, copy paste issue when I copied the changes out from the VMR. Thanks.

File: eng/targets/ResolveReferences.targets [hunk 3]
```diff
@@ -227,10 +227,6 @@
       Condition=" '@(PackageReference->HasMetadata('Private')->Count())' != '0' "
       Code="BUILD004"
       Text="%25(Private) metadata should not be applied to the %(Identity) package reference. Did you mean %25(PrivateAssets)?" />
-    <Warning
```
Comment (wtgodbe) [inline eng/targets/ResolveReferences.targets:230]: Is this just not true anymore?

File: eng/Npm.Workspace.FunctionalTests.nodeproj (hunk 2 above)
Comment (ViktorHofer) [reply eng/Npm.Workspace.FunctionalTests.nodeproj:1]: This is just correctness as without a DefaultTargets, the very first target gets built which is the Restore target in this project file. Not necessary for this to be green but the right thing to do.

File: eng/targets/ResolveReferences.targets (hunk 3 above)
Comment (ViktorHofer) [reply eng/targets/ResolveReferences.targets:230]: That warning was wrong to begin with. The `PrivateAssets` metadata is supported on both ProjectReference and PackageReference. This started failing when building with desktop msbuild.

**Comment by wtgodbe** [comment]
Is there any reason we can't update the Windows build in ci/ci-public.yml to use the new one-pass build.cmd invocation you mentioned on Teams?

**Review by wtgodbe — APPROVED** [review]
This is great, thanks!

**Pull request merged**